src/IMangrove.sol

```diff
@@ -1,27 +1,52 @@
 // SPDX-License-Identifier: Unlicense
 // This file was manually adapted from a file generated by abi-to-sol. It must
-// be kept up-to-date with the actual Mangrove interface. Fully automatic
-// generation is not yet possible due to user-generated types in the external
-// interface lost in the abi generation.
+// be kept up-to-date with the actual Mangrove interface.
 
 pragma solidity >=0.7.0 <0.9.0;
 
 import "mgv_src/core/MgvLib.sol";
 
+///@title Interface for the Mangrove contract.
 interface IMangrove is HasMgvEvents {
   // # Permit functions
 
+  ///@notice See {IERC20Permit-DOMAIN_SEPARATOR}.
+  ///@return the domain separator.
   function DOMAIN_SEPARATOR() external view returns (bytes32);
 
+  ///@notice See {IERC20Permit-PERMIT_TYPEHASH}.
+  ///@return The permit type hash.
   function PERMIT_TYPEHASH() external pure returns (bytes32);
 
+  ///@notice approves the spender to spend the amount of tokens on behalf of the caller.
+  ///@param outbound_tkn The address of the outbound token.
+  ///@param inbound_tkn The address of the inbound token.
+  ///@param spender The address of the spender.
+  ///@param value The amount of tokens to approve.
+  ///@return true if the approval succeeded; always true.
   function approve(address outbound_tkn, address inbound_tkn, address spender, uint value) external returns (bool);
 
+  ///@notice returns the allowance of the spender to spend tokens on behalf of the owner.
+  ///@param outbound_tkn The address of the (maker) outbound token.
+  ///@param inbound_tkn The address of the (maker) inbound token.
```

Comment:
Lack of cross referencing and terminology definitions make it a bit of a mess. Because is it known here that outbound is the maker's perspective?
I have now added (maker) to the other.